src/ProjectFileManager.cpp

```diff
@@ -1375,9 +1375,11 @@ void ReactOnMusicFileImport(
 
    if (!isFirstWaveTrack && isBeatsAndMeasures)
    {
+      auto &ph = ProjectHistory::Get(project);
       clip->SetRawAudioTempo(syncInfo.rawAudioTempo);
       clip->TrimQuarternotesFromRight(syncInfo.excessDurationInQuarternotes);
       clip->StretchBy(syncInfo.stretchMinimizingPowOfTwo);
+      ph.ModifyState(true);
```

Comment:
I, like you, would have thought this change is necessary now that I see it, but I just tried and undo/redo restores the stretch correctly _even without this change_ ...
I think this is because these lines are executed synchronously to file import, and later there is `   AddImportedTracks(fileName, std::move(newTracks));` which in turns pushes a new state in the history.

Reply:
It was unnecessary in the commit I based this on at first.  But it will be necessary in latest master, which delays those lines into the `CallAfter`.